### PR TITLE
[Google Blockly] Add CdoFieldToggle

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR adds `CdoFieldToggle`, a new customized field for Google Blockly labs. This field toggles between two different icons (for example a plus or minus icon) and triggers the`onClick` function which is passed via the constructor when the field is clicked.

There is an additional function `callback` and if it is passed in the constructor, it is called if `useDefaultIcon` is `false`, i.e., the second icon is displayed. An example of when this is needed is the `whenSpriteClicked` block which contains a flyout field which toggles between being displayed/hidden. If the block is saved when the flyout field is displayed, then when the block is reloaded, the flyout field would be displayed with`useDefaultIcon` assigned as `false`. 

## Links
[jira](https://codedotorg.atlassian.net/browse/SL-992)

## Testing story
This field was implemented and tested as part of @mikeharv's [branch](mike/spritelab-minitoolbox-blocks) which implemented the flyout block.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by mikeharv — APPROVED** [review]
Thanks for putting this field and PR together @fisher-alice! I've been using it extensively in my bigger [draft feature branch](https://github.com/code-dot-org/code-dot-org/compare/mike/spritelab-minitoolbox-blocks) for the mini-toolbox blocks and it's working great.

File: apps/src/blockly/addons/cdoFieldToggle.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import GoogleBlockly from 'blockly/core';
+
+export default class CdoFieldToggle extends GoogleBlockly.Field {
+  /**
+   * This is a customized field which the user clicks to toggle between two different states,
+   * for example, displaying or hiding the the block flyout.
+   *
+   * @param {Object} options - The options for constructing the class.
+   * @param {Function} options.onClick - The function that handles the field's editor.
+   * @param {SVGElement} options.icon1 - SVG <tspan> element - this is the icon that is initially displayed on the button.
```
Comment (molly-moen) [inline apps/src/blockly/addons/cdoFieldToggle.js]: do the names `icon1` and `icon2` come from blockly? If not could we make them more specific?

**Review by molly-moen — APPROVED** [review]
LGTM, just one comment/question on naming

File: apps/src/blockly/addons/cdoFieldToggle.js (hunk 1 above)
Comment (fisher-alice) [reply apps/src/blockly/addons/cdoFieldToggle.js]: They do not from mainline! I can rename. Thanks!

**Pull request merged**